source/extensions/filters/http/grpc_http1_bridge/http1_bridge_filter.cc

```diff
@@ -18,12 +20,32 @@ namespace Extensions {
 namespace HttpFilters {
 namespace GrpcHttp1Bridge {
 
+void Http1BridgeFilter::updateContentLength(Http::RequestHeaderMap& headers, uint64_t delta) {
+  uint64_t length;
+  if (absl::SimpleAtoi(headers.getContentLengthValue(), &length)) {
+    if (length <= std::numeric_limits<uint64_t>::max() - delta) {
+      headers.setContentLength(length + delta);
+    } else {
+      ENVOY_LOG(debug, "Unabled to update content lenght for protobuf to gRPC upgrade (overflow)");
```

Comment:
The one thing I'm wondering is what should happen in this case, because the content-length won't be equal to the data's length. Not an expert here so not sure how this is impacting the rest of the processing, but one option to consider is removing content-length.

```suggestion
      ENVOY_LOG(debug, "Unable to update content length for protobuf to gRPC upgrade (overflow)");
```

Reply:
I was thinking about your comment and there's actually one more nuance here that I'm not sure how best to handle. Even though `setContentLength` receives a `uint64_t` the gRPC header uses 32 bits for representing the message length, so one could  also argue that our overflow check should be against `std::numeric_limits<uin32_t>` instead.

Alternatively we could also skip updating the header entirely, as I don't think gRPC servers are expected to use it anyway.

What are your thoughts on this? Feedback is highly appreciated :sweat_smile:

Reply:
Upon a closer look at the gRPC-java implementation it looks like content-length is [explicitly discarded](https://github.com/grpc/grpc-java/search?q=CONTENT_LENGTH_KEY). With that in mind I think it makes more sense to leave it unchanged and let the gRPC server implementation handle it accordingly.

I'll go ahead and remove this update entirely, but let me know in case I'm missing something.